packages/plugins/src/plugins/balancer2/getPoolPositionsFetcher.ts

```diff
@@ -0,0 +1,190 @@
+          value: summedValue,
+          yields: [],
+          rewardAssets: [],
+          rewardAssetsValue: null,
+        };
+        poolLiquidities.push(liquidity);
+        continue;
+      }
+
+      // Farming Position
+      if (stackedBalance.stakingType === 'GAUGE') {
+        const { gaugeAddress } = poolPosition.staking.gauge;
+
+        const [balRewardsRaw, rewardTokensRaw] = await Promise.all([
+          getOwnerBalRewards(networkId, owner, gaugeAddress),
+          getOwnerGaugeRewards(networkId, owner, gaugeAddress),
+        ]);
+
+        const allRewardTokensWithBalance = [
+          balRewardsRaw,
+          ...rewardTokensRaw,
+        ].filter((token) => token.balance.isGreaterThan(0));
+
+        const allRewardTokenWithBalanceAndPrice = await Promise.all(
+          allRewardTokensWithBalance.map(async (token) => {
+            const tokenPrice = await cache.getTokenPrice(
+              token.address,
+              networkId
+            );
+            return {
+              ...token,
+              tokenPrice,
+            };
+          })
+        );
+
+        const rewardAssets: PortfolioAssetToken[] =
+          allRewardTokenWithBalanceAndPrice.flatMap((rewardAsset) => {
+            // Could add a safety check here so if the token price doesn't exist we fetch the decimal from the contract?
+            const balance = rewardAsset.balance.dividedBy(
+              new BigNumber(10).pow(rewardAsset?.tokenPrice?.decimals || 18)
+            );
+            return tokenPriceToAssetTokens(
+              rewardAsset.address,
+              balance.toNumber(),
+              networkId,
+              rewardAsset.tokenPrice,
+              rewardAsset.tokenPrice?.price || undefined
+            );
+          });
+
+        const rewardAssetsValue = getUsdValueSum(
+          rewardAssets.map((a) => a.value)
+        );
+
+        const liquidity: PortfolioLiquidity = {
+          assets,
+          assetsValue: summedValue,
+          value: summedValue,
+          yields: [],
+          rewardAssets,
+          rewardAssetsValue,
+        };
+        gaugeLiquidities.push(liquidity);
+        continue;
+      }
+
+      // TODO suuport locked positions
+      // if (stackedBalance.stakingType === 'VEBAL') {
+      // This is not available on fraxtal
+      // need to query contract for unlock time
+      // }
+
+      // TODO aura staking - they appear under aura finance on debank?
+      // if (stackedBalance.stakingType === 'AURA') {
+      // need to query contract for unlock time
+      // }
+    }
+
+    const elements: PortfolioElementLiquidity[] = [];
+    if (poolLiquidities.length !== 0) {
+      const element: PortfolioElementLiquidity = {
+        networkId,
+        platformId,
+        label: 'LiquidityPool',
+        type: PortfolioElementType.liquidity,
+        data: {
+          liquidities: poolLiquidities,
+        },
+        value: getUsdValueSum(poolLiquidities.map((a) => a.value)),
```

Comment:
Did you consider:

```suggestion
        value: getUsdValueSumFromLiquitidies(poolLiquidities),
```
?

The `getUsdValueSumFromLiquitidies()` utility would know how to dig into an array of objects and we wouldn't need to copy-pasta this boring code on line 175 below (and in, I'm guessing, tens  of other locations).

What do you think ?